### DISCUSSION TITLE
Kudos

### DISCUSSION THREAD

**Discussion by JetForMe** [HexFiend/HexFiend#475] [General]
I've been using Hex Fiend on and off for many, many years. The last couple days I'm debugging some MODBUS problems, and I'll sometimes take a string of logged bytes and paste them into Hex Fiend to use the nice little value interpreter in the lower part of the window. Today I had some new strings that looked like this:

```
[01][37][00][00][00][06][52][03][04][B8][00][02]
<36><00><00><00><07><88><52><03><04><38><A3><D7><01>
```

Not expecting it to work, I pasted them into Hex Fiend and…it worked perfectly. Hex Fiend ignored the brackets and just cleanly interpreted the characters.

Such a simple little detail, yet so helpful.

I just wanted to say a big THANK YOU to whomever thought to implement it like that. Just outstanding.

**Comment by kainjow** [comment]
Nice! I didn't even know it did this, and looked into it. It works via the `char2hex()` function which returns -1 on an invalid character, which just gets ignored:
https://github.com/HexFiend/HexFiend/commit/99ed5a3d7d1b4b2cae4432650c0fa50406edfbf3#diff-a7ca4336c81d5c882bc8bf793a90de99b59c41fd4895383f768ccaf3c97d8a9eR279

**Reply by JetForMe** [reply]
Nice!

**Comment by JanX2** [comment]
@JetForMe 
A bit off topic, but I’m working Modbus stuff as well. I’ve written a prototype Obj-C framework and CLI app (still pretty rough) that allows you to describe a device in a plist file (registers, format, unit, named group, metadata) and convert that to human-readable output. 

Also preparing a proposal for adding two additional endianness formats to HexFiend: BE and LE with swapped/reversed two-byte/register groups. Some of my devices deliver data in this fashion. 

Interested in details? You can contact me via the email address I use for commits.

**Comment by aussiegeek** [comment]
Just adding my own kudos. I don't use HexFiend frequently, but it's damn handy when I need it. Also, still my all time favourite app icon
